lab28_miya/Startup.cs

```diff
@@ -0,0 +1,101 @@
+using lab28_miya.Data;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace lab28_miya
+{
+    public class Startup
+    {
+        string _testSecret = null;
+
+        public IConfiguration Configuration
+        {
+            get;
+        }
+
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+        // This method gets called by the runtime. Use this method to add services to the container.
+        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
+        public void ConfigureServices(IServiceCollection services)
+        {
+            _testSecret = Configuration["MySecret"];
+
+            //Require https
+            services.Configure<MvcOptions>(options =>
+            {
+                options.Filters.Add(new RequireHttpsAttribute());
+            });
+
+            services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+                .AddCookie("MyCookieLogin", options =>
+                options.AccessDeniedPath = new PathString("/Account/Forbidden/"));
+
+            services.AddAuthorization(options =>
+            {
+                //this is where the policy is created
+                options.AddPolicy("Admin Only", policy => policy.RequireRole("Administrator"));
+                options.AddPolicy("MinimumYearsInService", policy => policy.Requirements.Add(new MinimumYearsInService()));
+                options.AddPolicy("Field Work", policy => policy.RequireRole("CPS Agent"));
+                options.AddPolicy("Skill Set", policy => policy.RequireClaim("StartDate"));
+            }
+            );
+
+            services.AddIdentity<ApplicationUser, IdentityRole>()
+    .AddEntityFrameworkStores<ApplicationDbContext>()
+    .AddDefaultTokenProviders();
+
+            services.AddAuthentication().AddFacebook(facebookOptions =>
+            {
+                facebookOptions.AppId = Configuration["MyFacebookAppID"];
+                facebookOptions.AppSecret = Configuration["5c199174958579481067b406ff15c229"];
+            }
+            );
+
+            services.AddSingleton<IAuthorizationHandler, Has5Years>();
+
+            services.AddMvc();
+
+            services.AddDbContext<lab28_miyaContext>(options =>
+                    options.UseSqlServer(Configuration.GetConnectionString("lab28_miyaContext")));
+
+                services.AddDbContext<ApplicationDbContext>(options =>
+            options.UseSqlServer(Configuration.GetConnectionString("lab28_miyaContext")));
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            app.UseAuthentication();
+
```

Comment:
You will need to enforce HTTPS for OAuth to work. Th code looks like this:

            var options = new RewriteOptions()
                .AddRedirectToHttps();

            app.UseRewriter(options);

You can find more information about it here: https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl